emu/src/io_registers.rs

```diff
@@ -0,0 +1,342 @@
+        }
+    }
+
+    pub fn set_byte(&mut self, byte: u8, value: u8) {
+        match self.access {
+            IORegisterAccessControl::Write | IORegisterAccessControl::ReadWrite => {
+                self.value.set_byte(byte, value);
+            }
+
+            _ => panic!("Requested IO Register cannot be written."),
+        }
+    }
+}
+
+#[allow(non_snake_case)]
+pub struct LCDRegisters {
+    // LCD Control
+    pub DISPCNT: IORegister,
+    // Undocumented - Green Swap
+    pub GreenSwap: IORegister,
+    // General LCD Status (STAT, LYC)
+    pub DISPSTAT: IORegister,
+    // Vertical Counter (LY)
+    pub VCOUNT: IORegister,
+    // BG0 Control
+    pub BG0CNT: IORegister,
+    // BG1 Control
+    pub BG1CNT: IORegister,
+    // BG2 Control
+    pub BG2CNT: IORegister,
+    // BG3 Control
+    pub BG3CNT: IORegister,
+    // BG0 X-Offset
+    pub BG0HOFS: IORegister,
+    // BG0 Y_Offset
+    pub BG0VOFS: IORegister,
+    // BG1 X-Offset
+    pub BG1HOFS: IORegister,
+    // BG1 Y_Offset
+    pub BG1VOFS: IORegister,
+    // BG2 X-Offset
+    pub BG2HOFS: IORegister,
+    // BG2 Y_Offset
+    pub BG2VOFS: IORegister,
+    // BG3 X-Offset
+    pub BG3HOFS: IORegister,
+    // BG3 Y_Offset
+    pub BG3VOFS: IORegister,
+    // BG2 Rotation/Scaling Parameter A (dx)
+    pub BG2PA: IORegister,
+    // BG2 Rotation/Scaling Parameter B (dmx)
+    pub BG2PB: IORegister,
+    // BG2 Rotation/Scaling Parameter C (dy)
+    pub BG2PC: IORegister,
+    // BG2 Rotation/Scaling Parameter D (dmy)
+    pub BG2PD: IORegister,
+    // BG2 Reference Point X-Coordinate
+    pub BG2X: IORegister,
+    // BG2 Reference Point Y-Coordinate
+    pub BG2Y: IORegister,
+    // BG3 Rotation/Scaling Parameter A (dx)
+    pub BG3PA: IORegister,
+    // BG3 Rotation/Scaling Parameter B (dmx)
+    pub BG3PB: IORegister,
+    // BG3 Rotation/Scaling Parameter C (dy)
+    pub BG3PC: IORegister,
+    // BG3 Rotation/Scaling Parameter D (dmy)
+    pub BG3PD: IORegister,
+    // BG3 Reference Point X-Coordinate
+    pub BG3X: IORegister,
+    // BG3 Reference Point Y-Coordinate
+    pub BG3Y: IORegister,
+    // Window 0 Horizontal Dimensions
+    pub WIN0H: IORegister,
+    // Window 1 Horizontal Dimensions
+    pub WIN1H: IORegister,
+    // Window 0 Vertical Dimensions
+    pub WIN0V: IORegister,
+    // Window 1 Vertical Dimensions
+    pub WIN1V: IORegister,
+    // Inside of Window 0 and 1
+    pub WININ: IORegister,
+    // Inside of OBJ Window & Outside of Windows
+    pub WINOUT: IORegister,
+    // Mosaic Size
+    pub MOSAIC: IORegister,
+    // Color Special Effects Selection
+    pub BLDCNT: IORegister,
+    // Alpha Blending Coefficients
+    pub BLDALPHA: IORegister,
+    // Brightness (Fade-In/Out) Coefficient
+    pub BLDY: IORegister,
+}
+
+impl Default for LCDRegisters {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl LCDRegisters {
+    pub const fn new() -> Self {
+        Self {
+            DISPCNT: IORegister {
```

Comment:
We can simplify this with something like `IORegister::with_access_control(Write)` and it creates something like you did

Reply:
Yup, thanks